### PR TITLE
Improve docstring for `baremodule`.

### PR DESCRIPTION
**Pull request by EricForgy**

### CODE REVIEW

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -126,8 +126,22 @@ kw"__init__"
 """
     baremodule
 
-`baremodule` declares a module that does not contain `using Base`
-or a definition of [`eval`](@ref Base.eval). It does still import `Core`.
+`baremodule` declares a module that does not contain `using Base` or a definition of
```
Comment (simeonschaub) [inline base/docs/basedocs.jl:129]: Trailing Whitespace:
```suggestion
`baremodule` declares a module that does not contain `using Base` or a definition of
```

**Comment by EricForgy** [comment]
Thanks for suggestion @simeonschaub

File: base/docs/basedocs.jl [hunk 2]
```diff
@@ -126,8 +126,22 @@ kw"__init__"
...
+```
+
+is equivalent to
+
+```julia
+baremodule Foo
+
+using Base
+
+eval(ex) = Core.eval(@__MODULE__, ex)
```
Comment (mcabbott) [inline base/docs/basedocs.jl:143]: Could each of these perhaps have an `end`, a bit less confusing? 
```suggestion
    using Base
    eval(ex) = Core.eval(@__MODULE__, ex)
end
```

File: base/docs/basedocs.jl [hunk 3]
```diff
@@ -126,8 +126,22 @@ kw"__init__"
 """
     baremodule
 
-`baremodule` declares a module that does not contain `using Base`
-or a definition of [`eval`](@ref Base.eval). It does still import `Core`.
+`baremodule` declares a module that does not contain `using Base` or a definition of
+[`eval`](@ref Base.eval). It does still import `Core`. In other words,
+
+```julia
+module Foo
```
Comment (mcabbott) [inline base/docs/basedocs.jl:133]:
```suggestion
module Foo end
```

File: base/docs/basedocs.jl [hunk 4]
```diff
@@ -126,8 +126,22 @@ kw"__init__"
...
+```
+
+is equivalent to
+
+```julia
+baremodule Foo
+
+using Base
+
```
Comment (KristofferC) [inline base/docs/basedocs.jl:142]: I think there is also an `include` method being defined.

**Comment by simeonschaub** [comment]
Fun fact: I just discovered this is already documented [here](https://docs.julialang.org/en/v1/manual/modules/#Default-top-level-definitions-and-bare-modules-1), but I don't think it would hurt, to have this in the do string as well.

**Comment by EricForgy** [comment]
Thanks guys 👍 

Good catch @simeonschaub. I went ahead and updated it to be the same as the other section of docs (including the `end`) 😊

**Pull request merged**